libpimeval/src/pimResMgr.cpp

```diff
@@ -102,6 +102,9 @@ pimResMgr::~pimResMgr()
 }
 
 //! @brief  Alloc a PIM object
+//!         Expect bitsPerElement to be smaller than or equal to the width of dataType
+//!         For V layout, allocate bitsPerElement rows
+//!         For H layout, pad element to the width of dataType
 PimObjId
 pimResMgr::pimAlloc(PimAllocEnum allocType, uint64_t numElements, unsigned bitsPerElement, PimDataType dataType)
```

Comment:
Do we actually need `bitsPerElement`? Isn't it redundant? We can derive bits per element from the datatype.

Reply:
This is especially needed for V layout architectures. We support a set of common data types, but an application may just use fewer bits of it. For example, we can allocate an INT32 object but only use 20 bits of it. In this case, this bitsPerElement is essential for performance modeling.

Reply:
Then we should introduce INT20. I don't think this is a good design at all.

Reply:
I think we've discussed the following in emails. We need to support pimAllocAssociate with different data types, and PIM operations with mixed data types. For example, we may allocate an INT32 object first, then associate an INT8 or BOOL object with it. In this case, we would like to do element padding and align elements with different types. Each element will have a data width and a padded width. The code changes here are for supporting such functionality. We can remove bitsPerElements from the functional API later for simplicity, but I don't think code design is bad here